docs/useCanAccessCallback.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: default
+title: "useCanAccessCallback"
+---
+
+# `useCanAccessCallback`
+
+This hook returns a callback to call the `authProvider.canAccess()` method for a provided resource and action (and optionally a record).
+
+## Usage
+
+`useCanAccessCallback` returns an async function that takes an object `{ action, resource, record }` as argument. This function returns an object with the following properties:
+
+- `canAccess`: a boolean determining whether the user has access to the resource.
+- `error`: the error that might have occurred.
+
+```jsx
+import { useCanAccessCallback, useRecordContext, useDataProvider, Button } from 'react-admin';
+
+const ResetViewsButton = () => {
```

Comment:
This example isn't great: if the user cannot reset the views, they shouldn't see the button in the first place.

Reply:
Any suggestion?

Reply:
- A click handler that links to the record edit view if the user has the permission, or to the show view otherwise
- a custom exporter